infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -456,6 +456,7 @@ public void shutdown() {
         executor.shutdown();
```

Comment:
This doesn't wait for tasks to finish executing which can introduce a race that can cause a file descriptor leak.

When `clearDataFiles`  interleaves with any call to `getSegmentHandleForAddress` (i.e. through read/inspectAddresses).

Consider the following execution. A segment is closed, a read comes in and re-opens the segment, then the segment files are deleted. 

Can this not happen?

Reply:
Sure, this is a problem.
Actually we shut down all our netty servers asynchronously, probably I should make all `shutdown` methods synchronous

Reply:
Fixed